models/articles-model.js

```diff
@@ -0,0 +1,10 @@
+const db = require("../db/connection");
+
+function fetchArticleById(articleId) {
+  const sqlQuery = `SELECT * FROM articles WHERE articles.article_id = $1`;
+  return db.query(sqlQuery, [articleId]).then(({ rows }) => {
+    return rows.length === 0 ? Promise.reject({ articleFound: 0 }) : rows[0];
```

Comment:
interesting use of Promise.reject! Generally, I would suggest sending a custom message - but I can see you've handled that in your error handler. Might just be something to keep in mind for later, as you may end up crossing wires later on :)